### ISSUE TITLE
How to keep original/old data when onLoadError happens

### ISSUE THREAD

**Issue by YangPolar** [wenzhixin/bootstrap-table#3129]
When load the data to bootstrap table, the first time is ok. data is listed into Table.
But if the second time onLoadError happens, the original data in table will be disappeared.

How can I keep the original data to not disappear when onLoadError happens?

**Comment by wenzhixin** [comment]
I think this is a strange feature, but you can use this workaround to solve your problem:
```
var savedData = [];
$('#table').boostrapTable({
  onLoadSuccess: function (res) {
    savedData = res;
    return res;
  }
  onLoadError: function () {
    $('#table').boostrapTable('load', savedData);
  }
});
```

**Comment by YangPolar** [comment]
Thank you very much.

2018-01-06 11:15 GMT+09:00 文翼 <notifications@github.com>:

**Comment by YangPolar** [comment]
Hi everyone

I have a question, we set the bootstrap to run by post way, if we click the
history back button on browser, the JSP is displayed by cache.
But the bootstrap is refresh it again by post way not by cache.

How to solve the history back button problem?

For example, my jsp page contains the bootstrap and one div and 3 links
text.
I clicked link1, the div text is 1. and bootstrap got result by select *
from table1 where a = 1;
And at this time I saved the session data to 1.
I clicked link2, the div text is 2. and bootstrap got result by select *
from table1 where a = 2;
And at this time I saved the session data to 2.
And then I clicked history back button, the div text is 1. But bootstrap
got result by select * from table1 where a = 2;
Because I set the session data is 2.

How to solve this problem?

Regards,

Yang

2018-01-09 10:32 GMT+09:00 陽立銘 <garywhitewolf@gmail.com>:

**Comment by YangPolar** [comment]
Hi everyone

I have a question, we set the bootstrap to run by post way, if we click the
history back button on browser, the JSP is displayed by cache.
But the bootstrap is refresh it again by post way not by cache.

How to solve the history back button problem?

For example, my jsp page contains the bootstrap and one div and 3 links
text.
I clicked link1, the div text is 1. and bootstrap got result by select *
from table1 where a = 1;
And at this time I saved the session data to 1.
I clicked link2, the div text is 2. and bootstrap got result by select *
from table1 where a = 2;
And at this time I saved the session data to 2.
And then I clicked history back button, the div text is 1. But bootstrap
got result by select * from table1 where a = 2;
Because I set the session data is 2.

How to solve this problem?

Regards,

Yang
